GeoLib/Polygon.cpp

```diff
@@ -30,6 +30,33 @@ Polygon::Polygon(const Polyline &ply, bool init) :
 		initialise ();
 }
 
+Polygon::Polygon(Polygon const& other)
+    : Polyline(other), _aabb(other._aabb)
+{
+	for (auto const* sub_polygon : other._simple_polygon_list) {
+		_simple_polygon_list.emplace_back(new Polygon(*sub_polygon));
+	}
+}
+
+Polygon& Polygon::operator=(Polygon const& other)
+{
+	if (other._ply_pnts != _ply_pnts) {
+		ERR(
+		    "Polygon::operator=: polygon other and this polygon are not based "
+		    "on the same point vector. Aborting");
+		std::abort();
+	}
+	if (other == *this)
+		return *this;
+
+	Polygon tmp(other);
+	std::swap(tmp._aabb, _aabb);
```

Comment:
This can be handled in another PR but the precedence is created here:
`warning: definition of implicit copy assignment operator for 'AABB' is deprecated because it has a user-declared copy constructor [-Wdeprecated]`
... requested in std::swap<GeoLib::AABB> at this line.